adaptris/interlok-workflow-rest-services
# INTERLOK-3252 Add the friendly name to the context via MDC.put

## Motivation

Since all the management components built by workflow-rest-services uses "interlok components" under the covers; we will get logging from those components. However, in certain situations we definitely don't want to that. e.g. health-check where a load balancer is using it to check that the instance is up every 5 seconds; you will get extra logging that might not be desired.

The proposal is to add a SLF4J MDC entry when each of these management components is fired so that additional filtering can be done.

## Modification

- Add the 'friendlyname' against the MDC key "ManagementComponent" -> this is effectively this.getClass().getSimpleName()
- Extend JettyMessageConsumer with a custom implementation that adds the context; and also add it to the associated onAM method.
- Since we have lombok enabled; get rid of boilerplate (which might improve coverage).
- Switch httpResponseService use to be metadata driven to avoid potential threading issues.
- Update README with information about each components MDC entry.

## Result

No change, unless they customise their log4j2.

## Testing

* Configure and build for healthcheck
* Add a log4j configuration (excerpt)
```
  <Appenders>
    <Console name="Console" target="SYSTEM_OUT">
      <PatternLayout>
        <Pattern>%-5p [%t] [%c{1.}.%M()] %m%n</Pattern>
      </PatternLayout>
    </Console>
    <Null name="NullAppender" />
    <Routing name="RoutingIgnoreHealthcheck">
      <Routes>
        
        <Script name="RoutingInit" language="JavaScript"><![CDATA[
              "LOG";
        ]]>
        </Script>
        <Route ref="Console" key="LOG"/>
        <Route ref="NullAppender" key="IGNORE"/>
      </Routes>
    </Routing>
  </Appenders>
  <Loggers>
    ....
    <Root level="TRACE">
      <AppenderRef ref="RoutingIgnoreHealthcheck"/>
    </Root>
  </Loggers>
```
This configuration will always force the logging to goto `LOG` (ie to console) so you will see something like : 

```
DEBUG [Thread-4] [c.a.r.WorkflowHealthCheckComponent.lambda$start$0()] [{}] WorkflowHealthCheckComponent component started.
TRACE [WorkflowHealthCheckComponent] [c.a.c.h.j.MetadataHeaderHandler.handleHeaders()] [{ManagementComponent=WorkflowHealthCheckComponent}] Adding Metadata [http.header.Accept: */*]
TRACE [WorkflowHealthCheckComponent] [c.a.c.h.j.MetadataHeaderHandler.handleHeaders()] [{ManagementComponent=WorkflowHealthCheckComponent}] Adding Metadata [http.header.User-Agent: curl/7.58.0]
TRACE [WorkflowHealthCheckComponent] [c.a.c.h.j.MetadataHeaderHandler.handleHeaders()] [{ManagementComponent=WorkflowHealthCheckComponent}] Adding Metadata [http.header.Host: localhost:8080]
```

Restart, but switch the logging so that the if condition statement is in play. Now you won't see the MetadataHandler logging; and healthcheck should be completely silent.
